node/src/embedding/embedding_function.ts

```diff
@@ -26,3 +26,9 @@ export interface EmbeddingFunction<T> {
      */
   embed: (data: T[]) => Promise<number[][]>
 }
+
+export function isEmbeddingFunction<T> (value: any): value is EmbeddingFunction<T> {
+  return Object.keys(value).length === 2 &&
```

Comment:
Not familar with how typescript works. But what if value is an object that do not have `sourceColumn` and `embed` fields, but other fields?